### PR TITLE
[redesign] Add md formatting guide modal

### LINKED ISSUE #1322: [redesign] Show formatting help for markdown editor
Add the "formatting help" button which triggers the display of the following modal:
image

### PR DESCRIPTION
**Pull request by thi4go**
closes #1322

### CODE REVIEW

File: src/componentsv2/ModalMDGuide.jsx [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+          <br />
+            return false
+          </pre>
+        </div>
+      </div>
+    </div>
+  </div> 
+);
+
+const styles = {
```
Comment (fernandoabolafio) [inline src/componentsv2/ModalMDGuide.jsx]: Please move the styles into classes under a `ModalMDGuide.module.css` file to keep the same pattern used on other components for v2.

File: src/componentsv2/ProposalForm/ProposalForm.jsx [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
 import MarkdownEditor from "src/componentsv2/MarkdownEditor";
 import FilesInput from "src/componentsv2/Files/Input";
 import AttachFileButton from "src/componentsv2/AttachFileButton";
+import ModalMDGuide from "src/componentsv2/ModalMDGuide";
 import { useProposalForm } from "./hooks";
 
 const ProposalForm = ({ initialValues, onSubmit, history, disableSubmit }) => {
+  const [mdGuideModal, setMdGuideModal] = useState(false);
+  const handleOpenMdGuideModal = () => setMdGuideModal(true);
+  const handleCloseMdGuideModal = () => setMdGuideModal(false);
```
Comment (fernandoabolafio) [inline src/componentsv2/ProposalForm/ProposalForm.jsx]: you can replace the 3 lines above by using the `useBooleanState` hook:

```js
const [showMDGuideModal, openMDGuideModal, closeMDGuideModal] = useBooleanState(false)
```

Comment (thi4go) [reply src/componentsv2/ProposalForm/ProposalForm.jsx]: Sweet! Missed that hook :sweat_smile:

**Comment by thi4go** [comment]
Thanks for the review mate @fernandoabolafio, review changes are pushed!

**Pull request merged**